api/v1beta1/azuremachine_webhook.go

```diff
@@ -206,6 +206,13 @@ func (mw *azureMachineWebhook) ValidateUpdate(ctx context.Context, oldObj, newOb
 		}
 	}
 
+	if err := webhookutils.ValidateImmutable(
+		field.NewPath("Spec", "CapacityReservationGroupID"),
```

Comment:
```suggestion
		field.NewPath("spec", "capacityReservationGroupID"),
```

Reply:
@arkadeepsen can you push this change? I'll give it another lgtm once that is done.

Reply:
@willie-yao Done.